packages/modules/src/SessionKeyManagerModule.ts

```diff
@@ -103,11 +111,14 @@ export class SessionKeyManagerModule extends BaseValidationModule {
         leafData.sessionKeyData,
       ]);
 
+      const generatedSessionId = leafData.preferredSessionId ?? generateRandomHex();
```

Comment:
should be `leafData?.preferredSessionId`, if preferredSessionId is not mandatory

Reply:
that's what it is checking. and leafData is mandatory